src/Microsoft.AspNetCore.Mvc.ApiExplorer/ApiBehaviorApiDescriptionProvider.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+
+namespace Microsoft.AspNetCore.Mvc.ApiExplorer
+{
+    public class ApiBehaviorApiDescriptionProvider : IApiDescriptionProvider
+    {
+        private readonly IModelMetadataProvider _modelMetadaProvider;
+
+        public ApiBehaviorApiDescriptionProvider(IModelMetadataProvider modelMetadataProvider)
+        {
+            _modelMetadaProvider = modelMetadataProvider;
+        }
+
+        /// <remarks>
+        /// The order is set to execute after the default provider.
+        /// </remarks>
+        public int Order => -1000 + 10;
+
+        public void OnProvidersExecuted(ApiDescriptionProviderContext context)
+        {
+        }
+
+        public void OnProvidersExecuting(ApiDescriptionProviderContext context)
+        {
+            foreach (var description in context.Results)
+            {
+                if (!AppliesTo(description))
+                {
+                    continue;
+                }
+
+                foreach (var responseType in CreateProblemResponseTypes(description))
+                {
+                    description.SupportedResponseTypes.Add(responseType);
+                }
+            }
+        }
+
+        public bool AppliesTo(ApiDescription description)
+        {
+            return description.ActionDescriptor.FilterDescriptors.Any(f => f.Filter is IApiBehaviorMetadata);
+        }
+
+        public IEnumerable<ApiResponseType> CreateProblemResponseTypes(ApiDescription description)
+        {
+            if (description.ActionDescriptor.Parameters.Any() || description.ActionDescriptor.BoundProperties.Any())
+            {
+                // For validation errors.
+                yield return CreateProblemResponse(StatusCodes.Status400BadRequest);
+
+                if (description.ActionDescriptor.Parameters.Any(p => p.Name.EndsWith("id", StringComparison.OrdinalIgnoreCase)))
```

Comment:
Probably need to harden this too. Here's what I have in the analyzer so far:

```C#
private static bool IsIdParameter(string name)
{
    // Check if the parameter is named "id" (e.g. int id) or ends in Id (e.g. personId)
    if (name == null || name.Length < 2)
    {
        return false;
    }

    if (string.Equals("id", name, StringComparison.Ordinal))
    {
        return true;
    }

    if (name.Length > 3 && name.EndsWith("Id", StringComparison.Ordinal) && char.IsLower(name[name.Length - 2]))
    {
        return true;
    }

    return false;
}
```

Reply:
Sure. I will also add some tests for the details of this.

Reply:
btw that should be a `name.Length >= 3` right?

Reply:
Yup